File: benchmarks/official/suites/matmul/matmul.js
const n = 256;
const size = n * n;
const a = new Float64Array(size);
const b = new Float64Array(size);
const c = new Float64Array(size);

for (let i = 0; i < size; i++) {
    const row = Math.floor(i / n);
    const col = i % n;
    a[i] = ((row * col) % 13) * 0.1;
    b[i] = ((row + col) % 17) * 0.1;
}

for (let r = 0; r < n; r++) {
    for (let col = 0; col < n; col++) {
        let sum = 0.0;
        for (let k = 0; k < n; k++) {
            sum += a[r * n + k] * b[k * n + col];
        }
        c[r * n + col] = sum;
    }
}

console.log(Math.floor(c[0]));
console.log(Math.floor(c[size - 1]));

File: benchmarks/official/suites/matmul/matmul.py
def main():
    n = 256
    size = n * n
    a = [0.0] * size
    b = [0.0] * size
    c = [0.0] * size

    for i in range(size):
        row = i // n
        col = i % n
        a[i] = ((row * col) % 13) * 0.1
        b[i] = ((row + col) % 17) * 0.1

    for r in range(n):
        for col in range(n):
            s = 0.0
            for k in range(n):
                s += a[r * n + k] * b[k * n + col]
            c[r * n + col] = s

    print(int(c[0]))
    print(int(c[size - 1]))

if __name__ == "__main__":
    main()
